source/apiVolontaria/volunteer/tests/tests_view_Cells.py

```diff
@@ -0,0 +1,418 @@
+import json
+
+from rest_framework import status
+from rest_framework.test import APIClient, APITestCase
+
+from django.urls import reverse
+
+from apiVolontaria.factories import UserFactory, AdminFactory
+from location.models import Address, StateProvince, Country
+from ..models import Cell
+
+
+class CellsTests(APITestCase):
+
+    def setUp(self):
+        self.client = APIClient()
+
+        self.user = UserFactory()
+        self.user.set_password('Test123!')
+        self.user.save()
+
+        self.admin = AdminFactory()
+        self.admin.set_password('Test123!')
+        self.admin.save()
+
+        self.random_country = Country.objects.create(
+            name="Random Country",
+            iso_code="RC",
+        )
+        self.random_state_province = StateProvince.objects.create(
+            name="Random State",
+            iso_code="RS",
+            country=self.random_country,
+        )
+        self.address = Address.objects.create(
+            address_line1='random address 1',
+            postal_code='RAN DOM',
+            city='random city',
+            state_province=self.random_state_province,
+            country=self.random_country,
+        )
+        self.cell = Cell.objects.create(
+            name='my cell',
+            address=self.address,
+        )
+
+    def test_create_new_cell_with_permission(self):
+        """
+        Ensure we can create a new cell if we have the permission.
+        The Address, Country and StateProvince do not exist in the DB.
```

Comment:
Really good test. Can we just add some assert on the top of this test to ensure the `Address`, `Country` and `StateProvince` doesn't exist ?

In the future some developers can change the `setup` function and this test can be touched. I just want to be sure these input are good or the test return an error.

Reply:
Wow good call!

Reply:
Done